### ISSUE TITLE
UX Improvement - Highlight Colours

### ISSUE THREAD

**Issue by Sam-Jayes** [firasdib/Regex101#2613]
Issue type: UX Improvement
## Issue Description:
When selecting/clicking on a group (capture or non capture group) the highlight colour doesnt stand out.

3 images attached: (capture group labelled **"Example"**)
-  No capture group selected
<img width="1276" height="295" alt="Image" src="https://github.com/user-attachments/assets/1b938102-5563-4686-8bac-1b2b37d38146" />

- Example Capture group selected (dark colour, not clear)
<img width="1278" height="287" alt="Image" src="https://github.com/user-attachments/assets/c3ad66cb-c50b-43c3-9663-b988ff3ea0d2" />

## Suggested Improvement:
Change background colour to a bright colour when group is selected (for both dark and light mode)

- Screenshot of how the colour change makes it clearer. **Bright Pipes** and capture group stands out making it easier to see different sections of the regex
<img width="1277" height="277" alt="Image" src="https://github.com/user-attachments/assets/b09d5360-ce9d-41db-b1fa-abad052daecc" />

**Comment by firasdib** [comment]
I have updated the way this works now, would you mind taking a look and see if its better? Thanks.

**Comment by Sam-Jayes** [comment]
@firasdib defiantly a bit better. Some highlights are still a bit too dark and don't stand out enough. I think this depends on what the base colour of the brackets are. e.g. The more Green brackets are good. The more Browny brackets, still blend in too much.

In my opinion the colour should stand out enough to tell the difference really easily.

I appreciate the time taken on this! Please let me know if you plan to update this further/want more feedback

**Comment by firasdib** [comment]
Thanks for checking it out @Sam-Jayes. I will see if I can tweak it even further :)

**Comment by SamJayes** [comment]
Hi @firasdib I see that in the latest update of regex101 colour schemes have been implemented as a PRO feature. I was wondering if this  is planned to be release at free tier or  to suggest that this is maybe moved. This is a nice to have feature that would make the UX in general a bit easier (as per the description of this ticket) but isnt something that i would subscribe to pro for.

**Comment by firasdib** [comment]
> Hi [@firasdib](https://github.com/firasdib) I see that in the latest update of regex101 colour schemes have been implemented as a PRO feature. I was wondering if this is planned to be release at free tier or to suggest that this is maybe moved. This is a nice to have feature that would make the UX in general a bit easier (as per the description of this ticket) but isnt something that i would subscribe to pro for.

Thanks for the feedback. I understand where you're coming from.

The colour schemes are currently intended to remain a Pro feature. Features like this are part of how I fund the continued development and maintenance of regex101, while keeping the core functionality available for free.

I've tried to keep Pro reasonably priced, and it includes a growing set of additional features beyond themes. If you use regex101 regularly, I'd encourage you to give Pro a try -- both for the extra features and as a way to support the ongoing work on the project.

That said, I appreciate the feedback on where the line between free and Pro should be drawn. It is something I will continue to evaluate as Pro evolves.

**Comment by SamJayes** [comment]
@firasdib I think the main driver for this comment is that I still find the highlight colour in the standard colour scheme still too dark (depending on the colour of the group that is selected) which makes it hard to work with large regex patterns - the reason for me opening the bug ticket in the first place. If you could try improve this in the base colour scheme that would be a great improvement for my UX.
That said, I totally understand keeping colour schemes as a pro feature and will see if I can convince work to allow me a pro subscription for the feature and as a way to support the development you do.

**Comment by firasdib** [comment]
I understand, and I will place an item in the backlog to revisit this.
